cirq/sim/wave_function.py

```diff
@@ -15,14 +15,26 @@
 
 import itertools
 
-from typing import Dict, Iterable, List, Optional, Sequence, Tuple, Type, Union
+from typing import Dict, Iterable, List, Optional, Sequence, Tuple, Type, Union, \
+    TYPE_CHECKING
 
 import abc
 import numpy as np
 
 from cirq import linalg, ops, value
 from cirq.sim import simulator
 
+if TYPE_CHECKING:
+    import cirq
+
+STATE_VECTOR_LIKE = Union[
+    # Full big-endian computational basis state index.
+    int,
+    # Per-qudit computational basis values.
+    Sequence[Union[int, bool]],
```

Comment:
Since bools are ints, we can just say int instead of the union.

Reply:
```
In [1]: isinstance(bool, int)                                                                                                   
Out[1]: False
```

Reply:
```
>>> isinstance(True, int)
True
```

Reply:
Hahaha, I made a dumb mistake and @cduck showed me the error in my ways. `isinstance(False, int)` returns true.

Reply:
I do think that there's some user benefit to listing bool explicitly, but done.

Reply:
yeah these type annotations are for the user so there's no harm in being explicit